python/tests/graph/test_input_graph.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+import itertools
+
+import numpy as np
+# Use this to create parameterized test cases
+from parameterized import parameterized
+
+from ..tests import PythonUnitTestCase
+from .base_test_generators import GenTestCases
+
+#========================================================================
+# Don't have to modify the content below
+
+_TEST_CASES_GENERATORS = []
```

Comment:
this is too complicated, and I do not see some reasons that would warrant such complexity. Please write some normal tests, with `parametrized` for example.